src/sparseml/transformers/utils/sparse_model.py

```diff
@@ -250,58 +258,43 @@ def text_generation_from_pretrained(
         :param kwargs: keyword arguments to pass through to the AutoModel call
         :return: the created model for text generation
         """
-        SparseAutoModel._check_tf(model_name_or_path)
-        if not kwargs:
-            kwargs = {}
-        kwargs["from_tf"] = False
-        delayed = False
-        if "state_dict" not in kwargs:
-            kwargs["state_dict"], delayed = SparseAutoModel._loadable_state_dict(
-                model_name_or_path
-            )
+        if config.model_type == "opt":
+            # TODO: Talk to Alex whether this pathway needs to be maintained
+            def skip(*args, **kwargs):
+                pass
 
-        kwargs["config"].is_decoder = True
-        kwargs["config"].use_past = False
+            torch.nn.init.kaiming_uniform_ = skip
+            torch.nn.init.uniform_ = skip
+            torch.nn.init.normal_ = skip
 
-        model = AutoModelForCausalLM.from_pretrained(
-            model_name_or_path,
-            **kwargs,
-        )
-        SparseAutoModel.log_model_load(model, model_name_or_path, model_type, delayed)
+            model = OPTForCausalLM.from_pretrained(
+                model_name_or_path,
+                torch_dtype=torch_dtype,
+                config=config,
+                trust_remote_code=trust_remote_code,
+                **kwargs,
+            )
+        else:
+            model = AutoModelForCausalLM.from_pretrained(
+                model_name_or_path,
+                torch_dtype=torch_dtype,
+                config=config,
+                trust_remote_code=trust_remote_code,
+                **kwargs,
+            )
 
-        return model
+        # TODO: Do we need to call eval() here? Why?
+        model.eval()
+        model.seqlen = sequence_length
 
-    @staticmethod
-    def text_generation_from_pretrained_distil(
```

Comment:
Removing this per @markurtz request from yesterday's meeting (@Satrat 's presentation of SparseML Fine-Tuning module)